pkg/monitortests/network/onpremhaproxy/monitortest.go

```diff
@@ -0,0 +1,293 @@
+				constructedIntervals = append(constructedIntervals,
+					monitorapi.NewInterval(monitorapi.SourceHaproxyMonitor, monitorapi.Info).
+						//Locator(availableConditionChange.Locator).
+						Locator(monitorapi.Locator{Keys: map[monitorapi.LocatorKey]string{
+							monitorapi.LocatorOnPremKubeapiUnreachableFromHaproxyKey: key,
+						}}).
+						Message(monitorapi.NewMessage().Reason(monitorapi.OnPremHaproxyDetectsDown).
+							Constructed(monitorapi.ConstructionOwnerOnPremHaproxy).
+							HumanMessage(fmt.Sprintf("Kubeapi on %s is detected dead by %s", availableConditionChange.Message.Annotations[monitorapi.AnnotationNode], availableConditionChange.Locator.Keys["node"]))).
+						Display().
+						Build(previousChangeTime, availableConditionChange.From),
+				)
+			}
+
+			previousChangeTime = availableConditionChange.From
+			lastAvailableStatus = availableConditionChange.Message.Annotations[monitorapi.AnnotationStatus]
+		}
+
+		//deletionTime := time.Now()
+		//deletedIntervals := monitorapi.Intervals(allChangesForTheNode).Filter(func(eventInterval monitorapi.Interval) bool {
+		//	return eventInterval.Message.Reason == monitorapi.APIServiceDeletedInAPI
+		//})
+		//if len(deletedIntervals) > 0 {
+		//	deletionTime = deletedIntervals[0].To
+		//}
+		//if len(lastAvailableStatus) > 0 {
+		//	reason := monitorapi.APIServiceUnavailable
+		//	intervalLevel := monitorapi.Error
+		//	if lastAvailableStatus == "True" {
+		//		reason = monitorapi.APIServiceAvailable
+		//		intervalLevel = monitorapi.Info
+		//	} else if lastAvailableStatus != "False" {
+		//		reason = monitorapi.APIServiceUnknown
+		//		intervalLevel = monitorapi.Warning
+		//	}
+		//	constructedIntervals = append(constructedIntervals,
+		//		monitorapi.NewInterval(monitorapi.SourceAPIServiceMonitor, intervalLevel).
+		//			Locator(apiserviceLocator).
+		//			Message(monitorapi.NewMessage().Reason(reason).
+		//				Constructed(monitorapi.ConstructionOwnerAPIServiceLifecycle).
+		//				HumanMessage(previousHumanMesage)).
+		//			Display().
+		//			Build(previousChangeTime, deletionTime),
+		//	)
+		//}
+	}
+
+	return constructedIntervals, nil
+}
+
+func (*operatorLogAnalyzer) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	leaseIntervals := finalIntervals.Filter(func(eventInterval monitorapi.Interval) bool {
+		if eventInterval.Message.Reason == monitorapi.OnPremHaproxyDetectsDown {
+			return true
+		}
+		return false
+	})
+
+	testName := fmt.Sprint("[Jira: Networking / On-Prem Host Networking] Haproxy must be able to reach kubeapi server")
+	success := &junitapi.JUnitTestCase{Name: testName}
+	somethingFailed := false
+
+	testNameToFailures := map[string][]string{}
+	for _, interval := range leaseIntervals {
+		if interval.Message.Reason == monitorapi.OnPremHaproxyDetectsDown {
+			//CHOCOBOMB(mko) Status change itself is NOT an indication of failure. It's here only for development
```

Comment:
Maybe we could mark the test as failed if all three apiservers are down at the same time? We might have to filter out the initial state since I believe that will always be the case when haproxy first starts, but if it happens a second time it could be considered a failure.

Reply:
Can we first merge a working version and afterwards work on tweaks and make it shiny? I already get a whole long wishlist of what it would be cool if that was doing, but as the first pass it is important that this test keeps running all the time

Reply:
Yes, I'm good with that. The biggest win here is having the HAProxy status in the timeline, so let's get that in.